integration/e2e_test.go

```diff
@@ -0,0 +1,214 @@
+
+	// test query provider
+	createdProvider := providers[0]
+	provider := f.QueryProvider(createdProvider.Owner.String())
+	require.Equal(t, createdProvider, provider)
+
+	// Run provider service
+	provProc, provHost := f.ProviderStart(keyFoo, provHost)
+	t.Logf("provider %q running as: %v", provHost, provProc.Pid)
+	defer func() { // shutdown provider
+		err := provProc.Stop(true)
+		require.NoError(t, err)
+		so, se, err := provProc.ReadAll()
+		require.NoError(t, err)
+		t.Logf("Provider StdOut:\n%s", so)
+		t.Logf("Provider StdErr:\n%s", se)
+	}()
+
+	// Apply SDL to provider
+	// Create deployment for `keyBar`
+	f.TxCreateDeployment(deploymentOvrclkApp, fmt.Sprintf("--from=%s", keyBar), "-y")
+	tests.WaitForNextNBlocksTM(1, f.Port)
+
+	// test query deployments
+	deployments, err := f.QueryDeployments()
+	require.NoError(t, err)
+	require.Len(t, deployments, 1, "Deployment Create Failed")
+	require.Equal(t, barAddr.String(), deployments[0].Deployment.DeploymentID.Owner.String())
+
+	// test query deployment
+	createdDep := deployments[0]
+	deployment := f.QueryDeployment(createdDep.Deployment.DeploymentID)
+	require.Equal(t, createdDep, deployment)
+
+	tests.WaitForNextNBlocksTM(1, f.Port)
+	orders, err := f.QueryOrders()
+	require.NoError(t, err)
+	require.Len(t, orders, 1, "Order creation failed")
+	require.Equal(t, barAddr.String(), orders[0].OrderID.Owner.String())
+
+	// Assert that there are no leases yet
+	leases, err := f.QueryLeases()
+	require.NoError(t, err)
+	require.Len(t, leases, 0, "no Leases should be created yet")
+
+	// Wait for then EndBlock to handle bidding and creating lease
+	tests.WaitForNextNBlocksTM(5, f.Port)
+
+	// Assert provider made bid
+	leases, err = f.QueryLeases()
+	require.NoError(t, err)
+	require.Len(t, leases, 1, "Lease should be created after bidding completes")
+	lease := leases[0]
+
+	// Provide manifest to provider service
+	f.SendManifest(lease, deploymentOvrclkApp)
+
+	t.Logf("hanging out for a bit...")
+	tests.WaitForNextNBlocksTM(10, f.Port)
+
+	// Assert provider launches app in kind
+	queryApp(t)
+
+	// Close deployment to clean up container
+	//  Teardown/cleanup
+	f.TxCloseDeployment(fmt.Sprintf("--from=%s --dseq=%v", keyBar, createdDep.Deployment.DeploymentID.DSeq), "-y")
+	tests.WaitForNextNBlocksTM(3, f.Port)
+}
+
+func TestK8sClient(t *testing.T) {
+	// create k8s client
+	ctx := context.Background()
+	host := "localhost"
+	ns := "lease" // this is an odd variable which by default is 'lease'
+	log := log.NewTMLogger(os.Stdout)
+	k8sClient, err := kube.NewClient(log, host, ns)
+	require.NoError(t, err)
+
+	// TODO: get the app's selected port from k8s Deployment info
+	k8sDeployments, err := k8sClient.Deployments(ctx)
+	require.NoError(t, err)
+	assert.NotEqual(t, k8sDeployments, 0)
+	k8sGroup := k8sDeployments[0].ManifestGroup()
+	for _, dep := range k8sDeployments {
+		t.Logf("deployment: %#v", dep)
+		for _, svc := range k8sGroup.Services {
+			t.Logf("service: %#v", svc)
+		}
+	}
```

Comment:
not testing anything?

fwiw the client is [already tested](https://github.com/ovrclk/akash/blob/faf26a8704ecf39730597254b9633d3da6542ca7/provider/cluster/kube/k8s_integration_test.go#L55-L62)